src/main/java/org/broadinstitute/hellbender/tools/walkers/conversion/GtfToBed.java

```diff
@@ -0,0 +1,288 @@
+
+        // store in hashmap with key as geneId
+        idToInfo.put(gtfFeature.getGeneId(), gtfInfo);
+    }
+
+    // stores the transcript ID and Interval info in hashmap
+    private void processTranscriptFeature(GencodeGtfFeature gtfFeature) {
+        //get interval and put the interval, type as transcript, and the name of the gene it's in
+        final Interval interval = new Interval(gtfFeature.getContig(), gtfFeature.getStart(), gtfFeature.getEnd());
+        final GtfInfo gtfInfo = new GtfInfo(interval, GtfInfo.Type.TRANSCRIPT, gtfFeature.getGeneName());
+
+        //store in hashmap with key as transcriptId
+        idToInfo.put(gtfFeature.getTranscriptId(), gtfInfo);
+
+        //update start/end of corresponding gene if needed
+        updateGeneStart(gtfFeature);
+        updateGeneEnd(gtfFeature);
+    }
+
+    // update the gene interval start position based on the transcript
+    private void updateGeneStart(GencodeGtfFeature gtfFeature) {
+        // get the start value of the gene
+        int geneStart = idToInfo.get(gtfFeature.getGeneId()).getStart();
+
+        // if the transcript start is less than the gene start
+        if (gtfFeature.getStart() < geneStart) {
+            // set the gene start to be the transcript start
+            geneStart = gtfFeature.getStart();
+            updateGeneInterval(gtfFeature, geneStart, idToInfo.get(gtfFeature.getGeneId()).getEnd());
+        }
+    }
+
+    // update the gene interval end position based on the transcript
+    private void updateGeneEnd(GencodeGtfFeature gtfFeature) {
+        // get the end value of the gene
+        int geneEnd = idToInfo.get(gtfFeature.getGeneId()).getEnd();
+
+        // if the transcript start is greater than the gene start
+        if (gtfFeature.getEnd() > geneEnd) {
+            // set the gene end to be the transcript end
+            geneEnd = gtfFeature.getEnd();
+            updateGeneInterval(gtfFeature, idToInfo.get(gtfFeature.getGeneId()).getStart(), geneEnd);
+        }
+    }
+
+    // updates an interval of the gene if it needs to be changed
+    private void updateGeneInterval(GencodeGtfFeature gtfFeature, int geneStart, int geneEnd) {
+        Interval geneInterval = new Interval(gtfFeature.getContig(), geneStart, geneEnd);
+        GtfInfo gtfGeneInfo = new GtfInfo(geneInterval, GtfInfo.Type.GENE, gtfFeature.getGeneName());
+        idToInfo.put(gtfFeature.getGeneId(), gtfGeneInfo);
+    }
+
+    // runs immediately after it has gone through each line of gtf (apply method)
+    @Override
+    public Object onTraversalSuccess() {
+        // get the user input dictionary
+        SAMSequenceDictionary sequenceDictionary = getSequenceDictionary(String.valueOf(this.sequenceDictionary));
+
+        // create linked hash map to store sorted values of idToInfo
+        LinkedHashMap<String, GtfInfo> karyotypeIdToInfo = getSortedMap(sequenceDictionary);
```

Comment:
If you need a map to be sorted you can use a TreeMap with the comparator you want.  It will keep itself sorted that way.

Reply:
I didn't use TreeMap because I need to sort by value (I am sorting by Interval), and it only allows sorting by Key.